### PR TITLE
feat: extend region leases in Metasrv

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Part of #1737 .

Please see our internal doc [here](https://greptime.feishu.cn/wiki/YJVTweW8biHdIQkdT3ncjPuSnMb#Gt1LdVNrvow0UYxhKHbcpxmIn3b).

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by WenyXu — APPROVED** [review]

File: src/meta-srv/src/handler/region_lease_handler.rs [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+        });
+
+        let mut region_leases = Vec::with_capacity(datanode_regions.len());
+        for (table_ref, local_regions) in datanode_regions {
+            let table_global_key = TableGlobalKey {
+                catalog_name: table_ref.catalog.to_string(),
+                schema_name: table_ref.schema.to_string(),
+                table_name: table_ref.table.to_string(),
+            };
+            let Some(table_global_value) = table_routes::get_table_global_value(&self.kv_store, &table_global_key).await? else { continue };
```
Comment (fengjiachun) [inline src/meta-srv/src/handler/region_lease_handler.rs:112]: I think we cat get all table_global_value in one batch, then we can  move this rpc call out of the loop.

Comment (fengjiachun) [reply src/meta-srv/src/handler/region_lease_handler.rs:112]: Even if optimized to `batch_get`, I still think there may be serious performance bottlenecks here, which will block metasrv from executing other logic. We can leave a todo here and we should solve it as soon as possible.

File: src/catalog/src/remote/region_alive_keeper.rs [hunk 2]
```diff
@@ -107,6 +118,53 @@ impl RegionAliveKeepers {
...
+                    .and_then(|x| x.try_into())
+                {
+                    Ok(x) => x,
+                    Err(e) => {
+                        error!(e; "");
+                        continue;
+                    }
+                };
+
+                let Some(keeper) = keepers.lock().await.get(&table_ident).cloned() else {
```
[the remark this answers is not in the document]
Comment (MichaelScofield) [reply src/catalog/src/remote/region_alive_keeper.rs:155]: The `RegionLease` is already grouped in tables. I think the lock is in the smallest granularity.

[the remark this answers is not in the document]
Comment (fengjiachun) [reply src/catalog/src/remote/region_alive_keeper.rs:155]: `keep_lived` should execute quickly, so small locks are not necessarily optimal, heartbeat responses are serial, concurrency here is not large, locks have overhead, if it is moved to the outer layer, only one call is needed and there is not much competition.

**Review by fengjiachun — APPROVED** [review]

**Pull request merged**